frontend/packages/ceph-storage-plugin/src/components/bucket-class/modals/edit-backingstore-modal.tsx

```diff
@@ -17,35 +17,87 @@ import {
 } from '@console/internal/components/utils';
 import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
 import { getName } from '@console/shared';
-import { NooBaaBucketClassModel, NooBaaBackingStoreModel } from '../../../models';
+import {
+  NooBaaBucketClassModel,
+  NooBaaBackingStoreModel,
+  NooBaaNamespaceStoreModel,
+} from '../../../models';
 import { getBackingStoreNames, getBackingStorePolicy } from '../../../utils/noobaa-utils';
 import BackingStoreSelection from '../backingstore-table';
 import {
   BackingStoreKind,
   K8sListResponse,
   BucketClassKind,
   PlacementPolicy,
+  NamespaceStoreKind,
 } from '../../../types';
 import './_bs-modal.scss';
+import { initialState, reducer } from '../state';
+import { NamespacePolicyType } from '../../../constants/bucket-class';
+import { SingleNamespaceStorePage } from '../wizard-pages/namespace-store-pages/single-namespace-store';
+import { CacheNamespaceStorePage } from '../wizard-pages/namespace-store-pages/cache-namespace-store';
+import { MultiNamespaceStorePage } from '../wizard-pages/namespace-store-pages/multi-namespace-store';
+import { validateDuration } from '../../../utils/bucket-class';
 
 const BucketClassEditModal = withHandlePromise<
   HandlePromiseProps & BucketClassEditModalProps & ModalComponentProps & CreateModalLauncherProps
 >((props) => {
   const { t } = useTranslation();
   const { bucketClass, inProgress, errorMessage, handlePromise, close, cancel } = props;
+  const isNamespaceType = !!bucketClass.spec?.namespacePolicy;
```

Comment:
Also check for namespace store guard here if in case a policy can be added from CLI.
Other wise looks good.

Reply:
added the flag here too .. thanks